omnibus/config/software/updater.rb

```diff
@@ -89,6 +89,7 @@
       "datadog-agent-process-exp.service.erb" => "datadog-agent-process-exp.service",
       "datadog-agent-security-exp.service.erb" => "datadog-agent-security-exp.service",
       "datadog-agent-sysprobe-exp.service.erb" => "datadog-agent-sysprobe-exp.service",
+      "datadog-updater.service-exp.erb" => "datadog-updater-exp.service",
```

Comment:
datadog-updater.service.erb is currently added by deb install script (see above), 
For the loadUnit to work you need to move it below + remove unit commands start/stop [from deb scripts ](https://github.com/DataDog/datadog-agent/tree/main/omnibus/package-scripts/updater-deb)